src/cpp/session/SessionModuleContext.cpp

```diff
@@ -95,13 +95,65 @@
 
 #include "session-config.h"
 
+#ifdef _WIN32
+# define kPathSeparator ';'
+#else
+# define kPathSeparator ':'
+#endif
+
 using namespace rstudio::core;
 using namespace boost::placeholders;
 
 namespace rstudio {
 namespace session {   
 namespace module_context {
 
+namespace {
+
+Error modifyPath(const core::FilePath& path, bool append)
+{
+   Error error;
+
+   std::string currentPath;
+   error = r::exec::RFunction("base:::Sys.getenv")
+         .addParam("PATH")
+         .call(&currentPath);
+
+   if (error)
+      return error;
+
+   std::string pathString = string_utils::utf8ToSystem(path.getAbsolutePath());
+
+#ifdef _WIN32
+   std::replace(pathString.begin(), pathString.end(), '/', '\\');
+#endif
+
+   std::string newPath = append
+         ? fmt::format("{}{}{}", currentPath, kPathSeparator, pathString)
+         : fmt::format("{}{}{}", pathString, kPathSeparator, currentPath);
+
+   error = r::exec::RFunction("base:::Sys.setenv")
```

Comment:
What's the point of doing it via r exec instead of using the `core::system::setenv` ? what are the differences here?

Reply:
This is mainly for a Windows oddity. I don't fully understand the details, but basically, environment variables set via `core::system::setenv()` won't be visible to the R session via `Sys.getenv()`. I think this might have something to do with R being built against a static copy of the libc runtime, and so it ends up with its own environment variable "block" which is separate from the one normally visible from our own session.

By using R's APIs for setting / getting environment variables, we avoid these sorts of issues.